waspc/data/Generator/templates/react-app/src/auth/login.js

```diff
@@ -9,15 +9,15 @@ export default async function login(username, password) {
     setAuthToken(response.data.token)
     // This isn't really neccessary because we remove all private queries after
     // logout, but we do it to be extra safe.
-    // 
+    //
     // For example, in future versions, users might be able to get to an SPA
     // login page while there's an active session. This code will prevent data
     // leaks in such cases.
     //
     // TODO(filip): We are currently removing all the queries, but we should
     // remove only non-public, user-dependent queries - public queries are
     // expected not to change in respect to the currently logged in user.
-    await removeQueries()
+    await invalidateAndRemoveQueries()
```

Comment:
By only removing the query, we removed the `Observer`s from all the components that used `user` object, and then the component didn't rerender probably. 

Invalidating and *then* removing the query, makes sure that the `Observer`s are recreated in the components and rerenders happen.

Reply:
I think this comment is worth adding to the code (inside the `invalidateAnndRemoveQueries` function). You can even link this explanation in there: https://github.com/wasp-lang/wasp/pull/595#discussion_r877364290

Reply:
Should we consider removing the `removeQueries` function since it's not used anywhere anymore? It will only confuse us in the future.

Reply:
I've updated the comment and removed `removeQueries` since we have no use for it right now and it's a potential footgun.